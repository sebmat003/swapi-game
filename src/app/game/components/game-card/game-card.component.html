<mat-card class="game-card" [class.winner]="winner">
  <mat-card-content class="d-flex flex-column justify-content-center align-items-center w-100 h-100">
    <h3>{{ person.name }}</h3>
    <app-icon [icon]="'person' + (index + 1)" [width]="200" [height]="200"></app-icon>
    <div class="mt-4">
      <p>Mass: {{ person.mass }}</p>
      <p>Height: {{ person.height }}</p>
      <p>Birth year: {{ person.birth_year }}</p>
      <p>Eye color: {{ person.eye_color }}</p>
      <p>Skin color: {{ person.skin_color }}</p>
    </div>
  </mat-card-content>
</mat-card>
