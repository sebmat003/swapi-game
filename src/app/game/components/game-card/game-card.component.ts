import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { Person } from "../../game/person.models";

@Component({
  selector: 'app-game-card',
  templateUrl: './game-card.component.html',
  styleUrls: ['./game-card.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class GameCardComponent {
  @Input() person!: Person;
  @Input() winner = false;
  @Input() index!: number;
}
