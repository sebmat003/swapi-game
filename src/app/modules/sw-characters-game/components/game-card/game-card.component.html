<mat-card
  [attr.data-cy]="'game-card'"
  class="game-card"
  [class.winner]="winner"
>
  <mat-card-content
    class="d-flex flex-column justify-content-center align-items-center w-100 h-100"
  >
    <h3>{{ person.name }}</h3>
    <app-icon
      [icon]="'person' + (index + 1)"
      [width]="200"
      [height]="200"
    ></app-icon>
    <div class="mt-4">
      <p>Gender {{ person.gender }}</p>
      <p>Mass: {{ person.mass }}</p>
      <p>Height: {{ person.height }}</p>
      <p>Birth year: {{ person.birthYear }}</p>
      <p>Eye color: {{ person.eyeColor }}</p>
      <p>Skin color: {{ person.skinColor }}</p>
      <p>Hair color: {{ person.hairColor }}</p>
    </div>
  </mat-card-content>
</mat-card>
