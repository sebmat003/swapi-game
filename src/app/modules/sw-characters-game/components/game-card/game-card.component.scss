.game-card {
  min-height: 500px;
  width: 100%;
  box-shadow: 1px 5px 24px -3px rgba(200, 200, 200, 1);
  border-radius: 1rem;

  &.winner {
    border: 3px solid green;
  }
}
