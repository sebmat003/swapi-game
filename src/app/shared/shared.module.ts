import { NgModule } from '@angular/core';
import { CommonModule } from "@angular/common";
import { LoaderComponent } from "./components/loader/loader.component";
import { IconComponent } from './components/icon/icon.component';
import { ReactiveFormsModule } from "@angular/forms";
import { MatCardModule } from "@angular/material/card";
import { MatSelectModule } from "@angular/material/select";

export const COMPONENTS = [
  LoaderComponent,
  IconComponent
];
export const MODULES = [
  CommonModule,
  ReactiveFormsModule,
  MatCardModule,
  MatSelectModule,
];

@NgModule({
  declarations: [COMPONENTS],
  imports: [MODULES],
  exports: [COMPONENTS, MODULES]
})
export class SharedModule {
}
