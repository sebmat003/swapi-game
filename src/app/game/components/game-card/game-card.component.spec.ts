import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { GameCardComponent } from "./game-card.component";
import { createTestObjects } from "../../testing/game-testing.data";
import { SharedModule } from "../../../shared/shared.module";

const {person} = createTestObjects();

describe('GameCardComponent', () => {
  let spectator: Spectator<GameCardComponent>;
  const createComponent = createComponentFactory({
    component: GameCardComponent,
    imports: [SharedModule],
    detectChanges: false
  });

  beforeEach(() => {
    spectator = createComponent();

    spectator.setInput('person', person);
    spectator.setInput('index', 0);
  });

  it('Should match snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('Should set border to green if winner', () => {
    spectator.setInput('winner', true);

    expect(spectator.query('.game-card')).toHaveClass('winner')
  });
});
