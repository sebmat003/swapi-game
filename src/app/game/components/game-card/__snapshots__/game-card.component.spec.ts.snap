// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCardComponent Should match snapshot 1`] = `
<app-game-card
  index="0"
  person={[Function Object]}
  winner="false"
>
  <mat-card
    class="mat-mdc-card mdc-card game-card"
  >
    <mat-card-content
      class="mat-mdc-card-content d-flex flex-column justify-content-center align-items-center w-100 h-100"
    >
      <h3>
        Anakin Skywalker
      </h3>
      <app-icon>
        <svg
          height="200"
          width="200"
        >
          <use
            xlink:href="assets/icons/icons.svg#person1"
          />
        </svg>
      </app-icon>
      <div
        class="mt-4"
      >
        <p>
          Gender male
        </p>
        <p>
          Mass: 100
        </p>
        <p>
          Height: 100
        </p>
        <p>
          Birth year: 2002
        </p>
        <p>
          Eye color: blue
        </p>
        <p>
          Skin color: white
        </p>
        <p>
          Hair color: gray
        </p>
      </div>
    </mat-card-content>
  </mat-card>
</app-game-card>
`;
